build/use_ccache.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+set -e
+
+# Ccache (https://ccache.dev/) is used for compiler cache and could make
```

Comment:
https://github.com/ccache/ccache/wiki/CMake is another good pointer here.

Reply:
Good point.

BTW regarding how we trigger CMake, in other projects (https://github.com/pytorch/pytorch/blob/main/CMakeLists.txt, 
https://cs.android.com/android-llvm/toolchain/llvm-project/+/master:llvm/CMakeLists.txt;l=171?q=ccache%20file:cmakelist, 
https://cs.android.com/android/platform/superproject/main/+/main:external/jsoncpp/CMakeLists.txt;l=60?q=ccache%20file:cmakelist) usually they used -DCMAKE_C_COMPILER_LAUNCHER when triggering cmake. However, now we can export env var (  export CMAKE_C_COMPILER_LAUNCHER=ccache  export CMAKE_CXX_COMPILER_LAUNCHER=ccache) and it's a new feature so we can just use it. The wiki doesn't really apply to our use case then.

However, we can use the old way if that's easier to understand